packages/eslint-plugin/tests/rules/no-type-alias.test.ts

```diff
@@ -542,6 +482,66 @@ type KeyNames = keyof typeof SCALARS;
       code: 'type Foo = Record<string, number>;',
       options: [{ allowGenerics: 'always' }],
     },
+    {
+      code: 'type Foo = `a-${number}`;',
+      options: [{ allowTemplateLiterals: 'always' }],
```

Comment:
I see the `allowTemplateLiterals` option as a kind of more specific `allowAliases`: that if you want to just enable template literals, but not all aliases, you can use `allowTemplateLiterals`. But, if you do want to allow all aliases, you would use `allowAliases`.

This PR's change breaks that last bit (and would be a breaking change as a result): it makes `allowTemplateLiterals` always override `allowAliases`. I don't think we want that? Unless I'm misinterpreting (am I?), could you add some tests showing that enabling `allowAliases` allows aliased template literals?

Reply:
@JoshuaKGoldberg After looking over my code I realized that I had structured `allowTemplateLiterals` around `allowLiterals`. The behavior should be the same between the two and their relationship with `allowAliases`. Maybe I am mistaken about that, but I've just pushed up new tests that further show the similar behavior.

Reply:
Yeah what I'm looking for is a test roughly like this:

```js
// in the valid array
{
  code: 'type Foo = `foo-${number}`;',
  options: [{ allowAliases: 'always' }],
},
```

As in, if the user configures `allowAliases` and doesn't configure `allowTemplateLiterals`, only the value for `allowAliases` should be used.